lib/std/fs.zig

```diff
@@ -2190,10 +2190,18 @@ pub fn selfExePathAlloc(allocator: *Allocator) ![]u8 {
 /// TODO make the return type of this a null terminated pointer
 pub fn selfExePath(out_buffer: []u8) SelfExePathError![]u8 {
     if (is_darwin) {
-        var u32_len: u32 = @intCast(u32, math.min(out_buffer.len, math.maxInt(u32)));
-        const rc = std.c._NSGetExecutablePath(out_buffer.ptr, &u32_len);
+        // Note that _NSGetExecutablePath() will return "a path" to
+        // the executable not a "real path" to the executable.
+        var symlink_path_buf: [MAX_PATH_BYTES:0]u8 = undefined;
+        var u32_len: u32 = MAX_PATH_BYTES;
```

Comment:
Should this not be `MAX_PATH_BYTES + 1` to account for null?

Reply:
According to the docs, paths returned might even go over MAX_PATH_BYTES. I haven't addressed the issue as it requires changing a bunch of public signatures.

From the docs:

> Note that _NSGetExecutablePath() will return "a path" to
     the executable not a "real path" to the executable.  That is, the path
     may be a symbolic link and not the real file. With deep directories the
     total bufsize needed could be more than MAXPATHLEN.

Reply:
Oh sure, what I meant here was that we do pass in the buffer of effective length of `MAX_PATH_BYTES + 1`, so I guess it'd be good to inform the OS of that. In terms of handling of "deep directories", I guess an error is our best bet at least in the short term, as you've already done :+1:

Reply:
Ah I see what you mean. Makes sense.